draft-ietf-quic-qpack.md

```diff
@@ -456,31 +458,36 @@ referenced by a given relative index will change while interpreting instructions
 on the encoder stream.
 
 ~~~~~ drawing
-    +---+---------------+-----------+
-    | n |      ...      |   d + 1   |  Absolute Index
-    + - +---------------+ - - - - - +
-    | 0 |      ...      | n - d - 1 |  Relative Index
-    +---+---------------+-----------+
-      ^                       |
-      |                       V
-Insertion Point         Dropping Point
+      +---+---------------+-----------+
+      | n |      ...      |   d + 1   |  Absolute Index
+      + - +---------------+ - - - - - +
+      | 0 |      ...      | n - d - 1 |  Relative Index
+      +---+---------------+-----------+
+      ^                               |
+      |                               V
+Insertion Point                 Dropping Point
 
 n = count of entries inserted
 d = count of entries dropped
 ~~~~~
 {: title="Example Dynamic Table Indexing - Control Stream"}
 
 Because frames from request streams can be delivered out of order with
-instructions on the encoder stream, relative indices are relative to the Base
-Index at the beginning of the header block (see {{header-prefix}}). The Base
-Index is an absolute index. When interpreting the rest of the frame, the entry
-identified by Base Index has a relative index of zero.  The relative indices of
-entries do not change while interpreting headers on a request or push stream.
+instructions on the encoder stream, relative indices are relative to the Base at
+the beginning of the header block (see {{header-prefix}}). The Base is encoded
+as a value relative to the Insert Count, so it can be expressed as the number of
+entries inserted. The dynamic table entries up to the Base can be referenced.
```

Comment:
> The Base is encoded as a value relative to the Insert Count, so it can be expressed as the number entries inserted

I'm having a hard time understanding why you are trying to say here.  That Base in the number of entries inserted?  Or the encoding of Base is the number of entries inserted?

Reply:
Yeah, I tried to fix this without changing it too much and that turned out badly.  I've a rewrite for you.